.github/workflows/hil_test_trigger.yml

```diff
@@ -8,6 +8,10 @@ on:
     # Run workflow at the start of every day (12 AM UTC)
     - cron: "0 0 * * *"
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

Comment:
I'm curious about what the second half of the conditional does. I see this in the github docs and it seems like it will cancel all HIL workworkflows, even ones unrelated to this PR:

> ```
> concurrency:
>   group: ${{ github.ref }}
>   cancel-in-progress: true
> ```
> 
> Note that in this example, without defining a particular concurrency group, GitHub Actions will cancel any in-progress run of the job or workflow.
>